org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/BaseLaunchRequestJob.java

```diff
@@ -64,34 +77,52 @@ protected final void runToolingApiJob(IProgressMonitor monitor) {
         // start tracking progress
         monitor.beginTask(getJobTaskName(), IProgressMonitor.UNKNOWN);
 
-        ProcessDescription processDescription = createProcessDescription();
-        ProcessStreams processStreams = CorePlugin.processStreamsProvider().createProcessStreams(processDescription);
+        final ProcessDescription processDescription = createProcessDescription();
+        final ProcessStreams processStreams = CorePlugin.processStreamsProvider().createProcessStreams(processDescription);
 
         // fetch build environment
         List<ProgressListener> listeners = ImmutableList.<ProgressListener>of(DelegatingProgressListener.withFullOutput(monitor));
 
-        // apply the fixed attributes on the request o
-        BuildRequest<Void> request = createRequest();
-        FixedRequestAttributes fixedAttributes = getConfigurationAttributes().toFixedRequestAttributes();
-        fixedAttributes.apply(request);
-
-        // configure the request's transient attributes
-        request.standardOutput(processStreams.getOutput());
-        request.standardError(processStreams.getError());
-        request.standardInput(processStreams.getInput());
-        request.progressListeners(listeners.toArray(new ProgressListener[listeners.size()]));
-        request.cancellationToken(getToken());
-
-        // print the applied run configuration settings at the beginning of the console output
-        OutputStreamWriter writer = new OutputStreamWriter(processStreams.getConfiguration());
-        writeFixedRequestAttributes(fixedAttributes, writer, monitor);
-
-        // notify the listeners before executing the build launch request
-        Event event = new DefaultExecuteLaunchRequestEvent(processDescription, request);
-        CorePlugin.listenerRegistry().dispatch(event);
-
-        // launch the build
-        request.executeAndWait();
+        final FixedRequestAttributes attributes = getConfigurationAttributes().toFixedRequestAttributes();
+        ProjectConnection connection = null;
+        try {
+            // apply FixedRequestAttributes on connector and establish TAPI connection
+            GradleConnector connector = GradleConnector.newConnector().forProjectDirectory(attributes.getProjectDir());
```

Comment:
Ideally, Fixed/Transient attributes should have methods to apply themselves to `GradleConnector`/`ProjectConnection`/`ConfigurableLauncher`, so we don't need to repeat ourselves. We can do that in another PR though.